src/App.tsx

```diff
@@ -1,39 +1,65 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
+export class App extends React.Component {
+  timerId = 0;
+
+  state: State = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', (event: MouseEvent) => {
+      event.preventDefault();
+
+      this.setState({ hasClock: false });
+    });
```

Comment:
move callback into separete function
```suggestion
    document.addEventListener('contextmenu', this.hideClock);
```